privval/file.go

```diff
@@ -158,14 +173,26 @@ var _ types.PrivValidator = (*FilePV)(nil)
 
 // NewFilePV generates a new validator from the given key and paths.
 func NewFilePV(privKey crypto.PrivKey, keyFilePath, stateFilePath string) *FilePV {
+	keyFile, _, err := fileAndContents(keyFilePath)
+	if err != nil {
+		panic(err)
+	}
+	stateFile, _, err := fileAndContents(stateFilePath)
+	if err != nil {
+		keyFile.Close()
+		panic(err)
```

Comment:
We should just return an error here instead. The caller of this returns an error in its function signature anyway, so this is fine.
https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#dont-panic

Reply:
The signature returns (*FilePV)
![Screen Shot 2021-12-08 at 5 14 07 PM](https://user-images.githubusercontent.com/4898263/145316445-68f0330d-367d-43de-b840-6f3f3afbcf9c.png)

It is impossible to return an error without breaking these APIs and having to go change all the code in the universe that uses Tendermint. I don't want to break APIs in a change that deals with preventing open+close multiple times. Plus this is initiation, it is okay to panic in these cases instead of being silently swallowed up.